src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPDocumentMappingProvider.cs

```diff
@@ -202,6 +223,53 @@ public async override Task<WorkspaceEdit> RemapWorkspaceEditAsync(WorkspaceEdit
             return workspaceEdit;
         }
 
+        private bool TryGetDocumentChanges(WorkspaceEdit workspaceEdit, out TextDocumentEdit[] documentEdits, out SumType<CreateFile, RenameFile, DeleteFile>[] documentNonEditChanges)
+        {
+            documentEdits = Array.Empty<TextDocumentEdit>();
+            documentNonEditChanges = Array.Empty<SumType<CreateFile, RenameFile, DeleteFile>>();
+
+            if (workspaceEdit.DocumentChanges?.Value is TextDocumentEdit[] documentEditArray)
+            {
+                documentEdits = documentEditArray;
+                return true;
+            }
+
+            if (workspaceEdit.DocumentChanges?.Value is SumType<TextDocumentEdit, CreateFile, RenameFile, DeleteFile>[] sumTypeArray)
```

Comment:
>sumTypeArray [](start = 120, length = 12)

Is the order of the TextDocumentEdit wrt the other edits in this file significant? I only ask because it looks like you separate those out and they end up going into the remappedEdits before any of the Create/Rename/Delete edits.